### DISCUSSION TITLE
IP Address

### DISCUSSION THREAD

**Discussion by djcrampton** [theyosh/TerrariumPI#681] [Q&A] [answered]
Hi Total novice here and totally new to python.. I have had the  Terranium Pi running for a couple of weeks really enjoying it!

I am just in the process of moving the setup out to its final position and was looking to open port forwarding to enable me to access remotely. Is this the recomended way to going about this?

However it looks like i am 'double NAT'd' and have network issues to sort as i had to add a further router to get WiFi  for the Pi where needed. But as I have messed up the IP address, after reboot the i cannot access the web interface to change this. 

Running the trouble shooting  in the FAQ i can see the IP is wrong. is there a way to manually edit this now or a full re-install? 

The error i see initially is:
OSError: [Errno 99] Cannot assign requested address: ('2

Any advice appriciated. 

Dave

**Comment by djcrampton** [comment]
I have perhaps made some progress following the password reset FAQ getting into the sqlite3 database I can see where I entered the incorrect IP, but cannot for the life of me edit it.

**Comment by theyosh** [comment, marked as answer]
Resetting the listening IP can be done this way:

1. Login with ssh to the Pi and locate the database. Should be `/home/pi/Terrariumpi/data`
2. `sqlite3 terrariumpi.db` this should open sqlite3 editor
3. `update Setting set value='0.0.0.0' where id='host';` Run this SQL command.
4. Ctrl+D to exit sqlite database.

Now you should have reset the host ip and if you restart TP4, it should be accessible.

In order to access it from the internet, you need to know your Pi IP and should make a port forward on your DSL modem/router.

**Reply by djcrampton** [reply]
Thanks that has worked. i was close with the SQL editing but not close enough!
